src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -2414,6 +2414,101 @@ public GCCondemnedReasons[] PerHeapCondemnedReasons
                 return _PerHeapCondemnedReasons;
             }
         }
+
+        public enum TimingType
+        {
+            /// <summary>
+            /// This field records the time spent for marking a type of GCHandle named HNDTYPE_SIZEDREF.
+            /// This happens only for blocking GC, and for majority of the time, this does not happen (because there are no sizedref handles)
+            /// so you get a 0 for blocking GC and a Null for background GC.
+            ///
+            /// Available for .NET 6+
+            /// </summary>
+            MarkSizedRef,
+
+            /// <summary>
+            /// This field records the time spent for marking roots (except objects pointed by sizedref handle and their descendents)
+            /// This happen for any type of GC, and it should not be zero for all GCs
+            ///
+            /// Available for .NET 6+
+            /// </summary>
+            MarkRoot,
+
+            /// <summary>
+            /// This field records the time spent for nulling out short weak references.
+            /// Short weak references should be null out if the only reason an object is alive is because of finalization (or dependent handle?)
+            /// This happens for any type of GC, and it should not be zero for all GCs.
```

Comment:
not dependent handles - those are strong handles. 

I would explain short and strong weak references together. 

GC will trace through strong references and any object found dead at this point, if they are targets of a short weak ref, that ref's target will be null-ed.

GC then promotes any finalizable objects that are dead at this point, making them live again. This is what MarkScanFinalization reports.

GC looks at the long weak references' targets and null out any whose targets are still dead at this point (ie, were not promoted due to finalization).